plugins/destination/snowflake/client/write.go

```diff
@@ -77,6 +84,63 @@ func (c *Client) WriteTableBatch(ctx context.Context, name string, msgs message.
 	return err
 }
 
+func (c *Client) mergeIntoTable(ctx context.Context, table *schema.Table, f *os.File) error {
+	sql := fmt.Sprintf(mergeIntoTable, table.Name, c.createColumnsList(table), escapePath(filepath.Base(f.Name())), createPrimaryKeyList(table), updateColumnsList(table), insertColumnsList(table))
+
+	if _, err := c.db.ExecContext(ctx, sql); err != nil {
+		return fmt.Errorf("failed to merge file into table: %s: %w", sql, err)
+	}
+	return nil
+}
+
+func (c *Client) createColumnsList(table *schema.Table) string {
+	// creates a string like:
+	// $1:COL1::TEXT as COL1, $1:COL2::NUMBER as COL2, $1:COL3::TIMESTAMP_TZ as COL3
+	columnString := ""
```

Comment:
Maybe create it as a known-length slice and strings.Join at the end? Still takes the same amount of lines, but less allocs possibly? Looks simpler too.

columns := make([]string, len(table.Columns)
in loop: coumns[i] = fmt.Sprintf...
return strings.Join(columns, ",")

Reply:
done, changed all of them as it was easier to read when I needed to change the ON AND separator